climada/util/coordinates.py

```diff
@@ -1821,6 +1821,160 @@ def apply_box(df_exp):
     }
     return raster_out, meta
 
+def reproject_raster_data(source, src_crs, src_transform, dst_crs=None, dst_transform=None,
+                      dst_shape=None, dst_resolution=None, dst_global_origin=None,
+                      dst_global_extent=None,
+                      dst_buffer=0, resampling=None, conserve=None,
+                      **kwargs):
+    """
+    Convenience wrapper function around rasterio.warp.reproject :
+    Reproject (up- or downsample) 2D np.ndarray to a grid using rasterio.
+    The destination (target) grid is defined from dst_transform and optionally
+    dst_resolution, dst_global_origin, and dst_crs.
+    This function ensures that reprojected data with the same dst_resolution and
+    dst_global_origins are on the same global grid, i.e., no offset between
+    destination grid points for different source grids that are projected to
+    the same target resolution.
+
+    Parameters
+    ----------
+    source : np.ndarray
+        2D-array containing the values to be reprojected
+    src_crs : CRS
+        Source CRS
+    src_transform : Affine
+        Source transform, contains info on grid, e.g.:
+            Affine(0.00833333333333333, 0.0, -18.175000000000068,
+                                 0.0, -0.00833333333333333, 43.79999999999993)
+    dst_crs : CRS, optional
+        Destination CRS. The default is src_crs.
+    dst_transform : Affine, optional
+        Destination transform. The default is src_transform.
```

Comment:
I don't think that `src_transform` is a good default value for `dst_transform`. By default, the value should be determined from the destination CRS and resolution, together with the source raster's extent or the `dst_global_origin`.